sdk/forms/src/main/java/com/klaviyo/forms/KlaviyoWebViewDelegate.kt

```diff
@@ -15,129 +14,243 @@ import androidx.core.content.ContextCompat.startActivity
-            it.addJavascriptInterface(this, IAF_BRIDGE_NAME)
+            val html = Registry.config.applicationContext
+                .assets
+                .open("InAppFormsTemplate.html")
+                .bufferedReader()
+                .use(BufferedReader::readText)
+                .replace("SDK_NAME", Registry.config.sdkName)
+                .replace("SDK_VERSION", Registry.config.sdkVersion)
+                .replace("BRIDGE_NAME", BRIDGE_NAME)
+                .replace("BRIDGE_HANDSHAKE", handShakeData)
+                .replace("KLAVIYO_JS_URL", klaviyoJsUrl.toString())
+
+            webView.loadDataWithBaseURL(Registry.config.baseUrl, html, MIME_TYPE, null, null)
+        }
+
+        handshakeTimer?.cancel()
+        handshakeTimer = Registry.clock.schedule(Registry.config.networkTimeout.toLong()) {
+            Registry.log.debug("IAF WebView Aborted: Timeout waiting for Klaviyo.js")
+            close()
         }
     }
 
+    /**
+     * When [DOCUMENT_START_SCRIPT] is unsupported, we must inject our script [onPageStarted] instead
+     */
+    override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) = view
+        ?.takeUnless { !LOAD_FROM_FILE || isFeatureSupported(DOCUMENT_START_SCRIPT) }
+        ?.let { webView ->
+            Registry.log.verbose("$DOCUMENT_START_SCRIPT Unsupported")
+            webView.evaluateJavascript(loadScript) { }
+        } ?: Unit
+
+    /**
+     * Called when loading a resource encounters http status code >= 400
+     */
     override fun onReceivedHttpError(
         view: WebView?,
         request: WebResourceRequest?,
         errorResponse: WebResourceResponse?
     ) {
-        Registry.log.error("HTTP Error: ${errorResponse?.statusCode} - ${request?.url}")
+        Registry.log.warning("HTTP Error: ${errorResponse?.statusCode} - ${request?.url}")
         super.onReceivedHttpError(view, request, errorResponse)
     }
 
-    override fun shouldInterceptRequest(
-        view: WebView?,
-        request: WebResourceRequest?
-    ): WebResourceResponse? {
-        Registry.log.debug("Request: ${request?.url}")
-        return super.shouldInterceptRequest(view, request)
-    }
-
+    /**
+     * Intercept page navigation and redirect to an external browser application
+     */
     override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
         if (request?.isForMainFrame == true) {
-            Registry.log.debug("Overriding external URL to browser: ${request.url}")
+            Registry.log.info("Redirect URL to external browser: ${request.url}")
             val intent = Intent(Intent.ACTION_VIEW).apply {
                 data = request.url
                 addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
             }
             activity?.let {
                 startActivity(it, intent, null)
             } ?: run {
-                Registry.log.error("Unable to launch external browser, null activity reference")
```

Comment:
`error` logs and above are still included in production builds, so I try to stay at warning or lower unless there's a compelling reason.